File: Projetos/ACBrLib/Demos/C#/eSocial/Import/Dinamico/MT/ACBreSocial.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;
using ACBrLib.Core;
using ACBrLib.Core.DFe;
using ACBrLib.eSocial;

namespace ACBrLib.eSocial
{
    /// <inheritdoc />
    public sealed partial class ACBreSocial : ACBrLibHandle
    {
        #region Constructors

        public ACBreSocial(string eArqConfig = "", string eChaveCrypt = "") : base(IsWindows ? "ACBreSocial64.dll" : "libacbresocial64.so",
                                                                                      IsWindows ? "ACBreSocial32.dll" : "libacbresocial32.so")
        {
            var inicializar = GetMethod<eSocial_Inicializar>();
            var ret = ExecuteMethod(() => inicializar(ref libHandle, ToUTF8(eArqConfig), ToUTF8(eChaveCrypt)));

            CheckResult(ret);

            Config = new ACBreSocialConfig(this);
        }

        #endregion Constructors

        #region Properties

        public string Nome
        {
            get
            {
                var bufferLen = BUFFER_LEN;
                var buffer = new StringBuilder(bufferLen);

                var method = GetMethod<eSocial_Nome>();
                var ret = ExecuteMethod(() => method(libHandle, buffer, ref bufferLen));

                CheckResult(ret);

                return ProcessResult(buffer, bufferLen);
            }
        }

        public string Versao
        {
            get
            {
                var bufferLen = BUFFER_LEN;
                var buffer = new StringBuilder(bufferLen);

                var method = GetMethod<eSocial_Versao>();
                var ret = ExecuteMethod(() => method(libHandle, buffer, ref bufferLen));

                CheckResult(ret);

                return ProcessResult(buffer, bufferLen);
            }
        }

        public ACBreSocialConfig Config { get; }

        #endregion Properties

        #region Metodos

        #region Ini

        public override void ConfigGravar(string eArqConfig = "")
        {
            var gravarIni = GetMethod<eSocial_ConfigGravar>();
            var ret = ExecuteMethod(() => gravarIni(libHandle, ToUTF8(eArqConfig)));

            CheckResult(ret);
        }

        public override void ImportarConfig(string eArqConfig)
        {
            var lerIni = GetMethod<eSocial_ConfigImportar>();
            var ret = ExecuteMethod(() => lerIni(libHandle, ToUTF8(eArqConfig)));

            CheckResult(ret);
        }

        public override string ExportarConfig()
        {
            var bufferLen = BUFFER_LEN;
            var buffer = new StringBuilder(bufferLen);

            var method = GetMethod<eSocial_ConfigExportar>();
            var ret = ExecuteMethod(() => method(libHandle, buffer, ref bufferLen));

            CheckResult(ret);

            return ProcessResult(buffer, bufferLen);
        }

        public override void ConfigLer(string eArqConfig = "")
        {
            var lerIni = GetMethod<eSocial_ConfigLer>();
            var ret = ExecuteMethod(() => lerIni(libHandle, ToUTF8(eArqConfig)));

            CheckResult(ret);
        }

        public override T ConfigLerValor<T>(ACBrSessao eSessao, string eChave)
        {
            var method = GetMethod<eSocial_ConfigLerValor>();

            var bufferLen = BUFFER_LEN;
            var pValue = new StringBuilder(bufferLen);
            var ret = ExecuteMethod(() => method(libHandle, ToUTF8(eSessao.ToString()), ToUTF8(eChave), pValue, ref bufferLen));
            CheckResult(ret);

            var value = ProcessResult(pValue, bufferLen);
            return ConvertValue<T>(value);
        }

        public override void ConfigGravarValor(ACBrSessao eSessao, string eChave, object value)
        {
            if (value == null) return;

            var method = GetMethod<eSocial_ConfigGravarValor>();
            var propValue = ConvertValue(value);

            var ret = ExecuteMethod(() => method(libHandle, ToUTF8(eSessao.ToString()), ToUTF8(eChave), ToUTF8(propValue)));
            CheckResult(ret);
        }

        #endregion Ini

        #region Diversos

        public void CriarEventoeSocial (string eArqIni)
        {
            var method = GetMethod<eSocial_CriarEventoeSocial>();
            var ret = ExecuteMethod(() => method(libHandle, ToUTF8(eArqIni)));

            CheckResult(ret);
        }

        public string EnviareSocial(int aGrupo)
        {
            var bufferLen = BUFFER_LEN;
            var buffer = new StringBuilder(bufferLen);

            var method = GetMethod<eSocial_EnviareSocial>();
            var ret = ExecuteMethod(() => method(libHandle, aGrupo, buffer, ref bufferLen));

            CheckResult(ret);

            return ProcessResult(buffer, bufferLen);
        }

        public string ConsultareSocial(string eProtocolo)
        {
            var bufferLen = BUFFER_LEN;
            var buffer = new StringBuilder(bufferLen);

            var method = GetMethod<eSocial_ConsultareSocial>();
            var ret = ExecuteMethod(() => method(libHandle, ToUTF8(eProtocolo), buffer, ref bufferLen));

            CheckResult(ret);

            return ProcessResult(buffer, bufferLen);
        }

        public void CriarEnviareSocial(string eArqIni, int aGrupo)
        {
            var method = GetMethod<eSocial_CriarEnviareSocial>();
            var ret = ExecuteMethod(() => method(libHandle, ToUTF8(eArqIni), aGrupo));

            CheckResult(ret);
        }

        public void LimpareSocial()
        {
            var method = GetMethod<eSocial_LimpareSocial>();
            var ret = ExecuteMethod(() => method(libHandle));

            CheckResult(ret);
        }

        public void CarregarXMLEventoeSocial(string eArquivoOuXML)
        {
            var method = GetMethod<eSocial_CarregarXMLEventoeSocial>();
            var ret = ExecuteMethod(() => method(libHandle, ToUTF8(eArquivoOuXML)));

            CheckResult(ret);
        }

        public void SetIDEmpregador(string aIdEmpregador)
        {
            var method = GetMethod<eSocial_SetIDEmpregador>();
            var ret = ExecuteMethod(() => method(libHandle, ToUTF8(aIdEmpregador)));

            CheckResult(ret);
        }

        public void SetIDTransmissor(string aIdTransmissor)
        {
            var method = GetMethod<eSocial_SetIDTransmissor>();
            var ret = ExecuteMethod(() => method(libHandle, ToUTF8(aIdTransmissor)));

            CheckResult(ret);
        }

        public void TipoEmpregador(int aTipoEmpregador)
        {
            var method = GetMethod<eSocial_SetTipoEmpregador>();
            var ret = ExecuteMethod(() => method(libHandle, aTipoEmpregador));

            CheckResult(ret);
        }

        public void SetVersao(string sVersao)
        {
            var method = GetMethod<eSocial_SetVersaoDF>();
            var ret = ExecuteMethod(() => method(libHandle, ToUTF8(sVersao)));

            CheckResult(ret);
        }

        public string ConsultaIdentificadoresEventosEmpregador(string aIdEmpregador, int aTipoEvento, DateTime aPeriodoApuracao)
        {
            var bufferLen = BUFFER_LEN;
            var buffer = new StringBuilder(bufferLen);

            var method = GetMethod<eSocial_ConsultaIdentificadoresEventosEmpregador>();
            var ret = ExecuteMethod(() => method(libHandle, ToUTF8(aIdEmpregador), aTipoEvento, aPeriodoApuracao, buffer, ref bufferLen));

            CheckResult(ret);

            return ProcessResult(buffer, bufferLen);
        }

        public string ConsultaIdentificadoresEventosTabela(string aIdEmpregador, int aTipoEvento, string aChave, DateTime aDataInicial, DateTime aDataFinal)
        {
            var bufferLen = BUFFER_LEN;
            var buffer = new StringBuilder(bufferLen);

            var method = GetMethod<eSocial_ConsultaIdentificadoresEventosTabela>();
            var ret = ExecuteMethod(() => method(libHandle, ToUTF8(aIdEmpregador), aTipoEvento, aChave, aDataInicial, aDataFinal, buffer, ref bufferLen));

            CheckResult(ret);

            return ProcessResult(buffer, bufferLen);
        }

        public string ConsultaIdentificadoresEventosTrabalhador(string aIdEmpregador, string aCPFTrabalhador, DateTime aDataInicial, DateTime aDataFinal)
        {
            var bufferLen = BUFFER_LEN;
            var buffer = new StringBuilder(bufferLen);

            var method = GetMethod<eSocial_ConsultaIdentificadoresEventosTrabalhador>();
            var ret = ExecuteMethod(() => method(libHandle, ToUTF8(aIdEmpregador), aCPFTrabalhador, aDataInicial, aDataFinal, buffer, ref bufferLen));

            CheckResult(ret);

            return ProcessResult(buffer, bufferLen);
        }

        public string DownloadEventos(string aIdEmpregador, string aCPFTrabalhador, DateTime aDataInicial, DateTime aDataFinal)
        {
            var bufferLen = BUFFER_LEN;
            var buffer = new StringBuilder(bufferLen);

            var method = GetMethod<eSocial_DownloadEventos>();
            var ret = ExecuteMethod(() => method(libHandle, ToUTF8(aIdEmpregador), aCPFTrabalhador, aDataInicial, aDataFinal, buffer, ref bufferLen));

            CheckResult(ret);

            return ProcessResult(buffer, bufferLen);
        }

        public InfoCertificado[] ObterCertificados()
        {
            var bufferLen = BUFFER_LEN;
            var buffer = new StringBuilder(bufferLen);

            var method = GetMethod<eSocial_ObterCertificados>();
            var ret = ExecuteMethod(() => method(libHandle, buffer, ref bufferLen));

            CheckResult(ret);

            var certificados = ProcessResult(buffer, bufferLen).Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries);
            return certificados.Length == 0 ? new InfoCertificado[0] : certificados.Select(x => new InfoCertificado(x)).ToArray();
        }

        public string OpenSSLInfo()
        {
            var bufferLen = BUFFER_LEN;
            var buffer = new StringBuilder(bufferLen);

            var method = GetMethod<eSocial_OpenSSLInfo>();
            var ret = ExecuteMethod(() => method(libHandle, buffer, ref bufferLen));

            CheckResult(ret);

            return ProcessResult(buffer, bufferLen);
        }

        #endregion Diversos

        #region Private Methods

        protected override void FinalizeLib()
        {
            var finalizar = GetMethod<eSocial_Finalizar>();
            var codRet = ExecuteMethod(() => finalizar(libHandle));
            CheckResult(codRet);
        }

        protected override string GetUltimoRetorno(int iniBufferLen = 0)
        {
            var bufferLen = iniBufferLen < 1 ? BUFFER_LEN : iniBufferLen;
            var buffer = new StringBuilder(bufferLen);
            var ultimoRetorno = GetMethod<eSocial_UltimoRetorno>();

            if (iniBufferLen < 1)
            {
                ExecuteMethod(() => ultimoRetorno(libHandle, buffer, ref bufferLen));
                if (bufferLen <= BUFFER_LEN) return FromUTF8(buffer);

                buffer.Capacity = bufferLen;
            }

            ExecuteMethod(() => ultimoRetorno(libHandle, buffer, ref bufferLen));
            return FromUTF8(buffer);
        }

        #endregion Private Methods

        #endregion Metodos
    }
}

File: Projetos/ACBrLib/Demos/Java/eSocial/Imports/ACBrLibeSocial/src/com/acbr/esocial/ACBreSocial.java
package com.acbr.esocial;

import com.acbr.ACBrLibBase;
import com.acbr.ACBrSessao;
import com.acbr.nfe.TipoPathNFe;
import com.sun.jna.Library;
import com.sun.jna.Native;
import com.sun.jna.Platform;
import com.sun.jna.ptr.IntByReference;

import java.io.File;
import java.nio.ByteBuffer;
import java.nio.file.Paths;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;

public final class ACBreSocial extends ACBrLibBase {

    private interface ACBreSocialLib extends Library {

        static String JNA_LIBRARY_NAME = LibraryLoader.getLibraryName();
        public final static ACBreSocialLib INSTANCE = LibraryLoader.getInstance();

        class LibraryLoader {

            private static String library = "";
            private static ACBreSocialLib instance = null;

            private static String getLibraryName() {
                if (library.isEmpty()) {
                    if (Platform.isWindows()) {
                        library = Platform.is64Bit() ? "ACBreSocial64" : "ACBreSocial32";
                    } else {
                        library = Platform.is64Bit() ? "acbresocial64" : "acbresocial32";
                    }
                }
                return library;
            }

            public static ACBreSocialLib getInstance() {
                if (instance == null) {
                    instance = (ACBreSocialLib) Native.synchronizedLibrary(
                            (Library) Native.load(JNA_LIBRARY_NAME, ACBreSocialLib.class));
                }

                return instance;
            }
        }

        int eSocial_Inicializar(String eArqConfig, String eChaveCrypt);

        int eSocial_Finalizar();

        int eSocial_Nome(ByteBuffer buffer, IntByReference bufferSize);

        int eSocial_Versao(ByteBuffer buffer, IntByReference bufferSize);

        int eSocial_UltimoRetorno(ByteBuffer buffer, IntByReference bufferSize);

        int eSocial_ConfigImportar(String eArqConfig);

        int eSocial_ConfigExportar(ByteBuffer buffer, IntByReference bufferSize);

        int eSocial_ConfigLer(String eArqConfig);

        int eSocial_ConfigGravar(String eArqConfig);

        int eSocial_ConfigLerValor(String eSessao, String eChave, ByteBuffer buffer, IntByReference bufferSize);

        int eSocial_ConfigGravarValor(String eSessao, String eChave, String valor);

        int eSocial_CriarEventoeSocial(String eArqIni);
        
        int eSocial_EnviareSocial(Integer aGrupo, ByteBuffer buffer, IntByReference bufferSize);
        
        int eSocial_ConsultareSocial(String eProtocolo, ByteBuffer buffer, IntByReference bufferSize);
    
        int eSocial_CriarEnviareSocial(String eArqIni, Integer aGrupo);
        
        int eSocial_LimpareSocial();
        
        int eSocial_CarregarXMLEventoeSocial(String eArquivoOuXML);
        
        int eSocial_SetIDEmpregador(String aIdEmpregador);
        
        int eSocial_SetIDTransmissor (String aIdTransmissor);
        
        int eSocial_SetTipoEmpregador(Integer aTipoEmpregador);
        
        int eSocial_SetVersaoDF(String sVersao);
        
        int eSocial_ConsultaIdentificadoresEventosEmpregador(String aIdEmpregador, Integer aTipoEvento, double aPeriodoApuracao, ByteBuffer buffer, IntByReference bufferSize);
    
        int eSocial_ConsultaIdentificadoresEventosTabela(String aIdEmpregador, Integer aTipoEvento, String aChave, double aDataInicial, double aDataFinal, ByteBuffer buffer, IntByReference bufferSize);
        
        int eSocial_ConsultaIdentificadoresEventosTrabalhador(String aIdEmpregador, String aCPFTrabalhador, double aDataInicial, double aDataFinal, ByteBuffer buffer, IntByReference bufferSize);
    
        int eSocial_DownloadEventos(String aIdEmpregador, String aCPFTrabalhador, double aDataInicial, double aDataFinal, ByteBuffer buffer, IntByReference bufferSize);
        
        int eSocial_ObterCertificados(ByteBuffer buffer, IntByReference bufferSize);
    }

    public ACBreSocial() throws Exception {
        File iniFile = Paths.get(System.getProperty("user.dir"), "ACBrLib.ini").toFile();
        if (!iniFile.exists()) {
            iniFile.createNewFile();
        }

        int ret = ACBreSocialLib.INSTANCE.eSocial_Inicializar(toUTF8(iniFile.getAbsolutePath()), toUTF8(""));
        checkResult(ret);
    }

    public ACBreSocial(String eArqConfig, String eChaveCrypt) throws Exception {
        int ret = ACBreSocialLib.INSTANCE.eSocial_Inicializar(toUTF8(eArqConfig), toUTF8(eChaveCrypt));
        checkResult(ret);
    }

    @Override
    protected void dispose() throws Exception {
        int ret = ACBreSocialLib.INSTANCE.eSocial_Finalizar();
        checkResult(ret);
    }

    public String nome() throws Exception {
        ByteBuffer buffer = ByteBuffer.allocate(STR_BUFFER_LEN);
        IntByReference bufferLen = new IntByReference(STR_BUFFER_LEN);

        int ret = ACBreSocialLib.INSTANCE.eSocial_Nome(buffer, bufferLen);
        checkResult(ret);

        return fromUTF8(buffer, bufferLen.getValue());
    }

    public String versao() throws Exception {
        ByteBuffer buffer = ByteBuffer.allocate(STR_BUFFER_LEN);
        IntByReference bufferLen = new IntByReference(STR_BUFFER_LEN);

        int ret = ACBreSocialLib.INSTANCE.eSocial_Versao(buffer, bufferLen);
        checkResult(ret);

        return fromUTF8(buffer, bufferLen.getValue());
    }

    public void configLer() throws Exception {
        configLer("");
    }

    public void configLer(String eArqConfig) throws Exception {
        int ret = ACBreSocialLib.INSTANCE.eSocial_ConfigLer(toUTF8(eArqConfig));
        checkResult(ret);
    }

    public void configGravar() throws Exception {
        configGravar("");
    }

    public void configGravar(String eArqConfig) throws Exception {
        int ret = ACBreSocialLib.INSTANCE.eSocial_ConfigGravar(toUTF8(eArqConfig));
        checkResult(ret);
    }

    @Override
    public String configLerValor(ACBrSessao eSessao, String eChave) throws Exception {
        ByteBuffer buffer = ByteBuffer.allocate(STR_BUFFER_LEN);
        IntByReference bufferLen = new IntByReference(STR_BUFFER_LEN);

        int ret = ACBreSocialLib.INSTANCE.eSocial_ConfigLerValor(toUTF8(eSessao.name()), toUTF8(eChave), buffer, bufferLen);
        checkResult(ret);

        return processResult(buffer, bufferLen);
    }

    @Override
    public void configGravarValor(ACBrSessao eSessao, String eChave, Object value) throws Exception {
        int ret = ACBreSocialLib.INSTANCE.eSocial_ConfigGravarValor(toUTF8(eSessao.name()), toUTF8(eChave), toUTF8(value.toString()));
        checkResult(ret);
    }


    public void ConfigImportar(String eArqConfig) throws Exception {

        int ret = ACBreSocialLib.INSTANCE.eSocial_ConfigImportar(eArqConfig);
        checkResult(ret);

    }

    public String ConfigExportar() throws Exception {

        ByteBuffer buffer = ByteBuffer.allocate(STR_BUFFER_LEN);
        IntByReference bufferLen = new IntByReference(STR_BUFFER_LEN);

        int ret = ACBreSocialLib.INSTANCE.eSocial_ConfigExportar(buffer, bufferLen);
        checkResult(ret);

        return processResult(buffer, bufferLen);

    }
    
    public void criarEventoeSocial(String eArqIni) throws Exception {
        int ret = ACBreSocialLib.INSTANCE.eSocial_CriarEventoeSocial(eArqIni);
        checkResult(ret);
    }
    
    public String enviareSocial(Integer aGrupo) throws Exception {
        
        ByteBuffer buffer = ByteBuffer.allocate(STR_BUFFER_LEN);
        IntByReference bufferLen = new IntByReference(STR_BUFFER_LEN);
        
        int ret = ACBreSocialLib.INSTANCE.eSocial_EnviareSocial(aGrupo, buffer, bufferLen);
        checkResult(ret);
        
        return processResult(buffer, bufferLen);
    }
    
    public String consultareSocial(String eProtocolo) throws Exception {
        
        ByteBuffer buffer = ByteBuffer.allocate(STR_BUFFER_LEN);
        IntByReference bufferLen = new IntByReference(STR_BUFFER_LEN);
        
        int ret = ACBreSocialLib.INSTANCE.eSocial_ConsultareSocial(eProtocolo, buffer, bufferLen);
        checkResult(ret);
        
        return processResult(buffer, bufferLen);
    }
    
    public void criarEnviareSocial(String eArqIni, Integer aGrupo) throws Exception {
        int ret = ACBreSocialLib.INSTANCE.eSocial_CriarEnviareSocial(eArqIni, aGrupo);
        checkResult(ret);
    }
    
    public void limpareSocial() throws Exception {
        int ret = ACBreSocialLib.INSTANCE.eSocial_LimpareSocial();
        checkResult(ret);
    }
    
    public void carregarXMLEventoeSocial(String eArquivoOuXML) throws Exception{
        int ret = ACBreSocialLib.INSTANCE.eSocial_CarregarXMLEventoeSocial(eArquivoOuXML);
        checkResult(ret);
    }
    
    public void setIDEmpregador(String aIdEmpregador) throws Exception {
        int ret = ACBreSocialLib.INSTANCE.eSocial_SetIDEmpregador(aIdEmpregador);
        checkResult(ret);
    }
    
    public void setIDTransmissor(String aIdTransmissor) throws Exception {
        int ret = ACBreSocialLib.INSTANCE.eSocial_SetIDTransmissor(aIdTransmissor);
        checkResult(ret);
    }
    
    public void setTipoEmpregador(Integer aTipoEmpregador) throws Exception {
        int ret = ACBreSocialLib.INSTANCE.eSocial_SetTipoEmpregador(aTipoEmpregador);
        checkResult(ret);
    }
    
    public void setVersaoDF(String sVersao) throws Exception {
        int ret = ACBreSocialLib.INSTANCE.eSocial_SetVersaoDF(sVersao);
        checkResult(ret);
    }
    
    public String consultaIdentificadoresEventosEmpregador(String aIdEmpregador, Integer aTipoEvento, double aPeriodoApuracao) throws Exception{
        
        ByteBuffer buffer = ByteBuffer.allocate(STR_BUFFER_LEN);
        IntByReference bufferLen = new IntByReference(STR_BUFFER_LEN);
        
        int ret = ACBreSocialLib.INSTANCE.eSocial_ConsultaIdentificadoresEventosEmpregador(aIdEmpregador, aTipoEvento, aPeriodoApuracao, buffer, bufferLen);
        checkResult(ret);
        
        return processResult(buffer, bufferLen);
    }
    
    public String consultaIdentificadoresEventosTabela(String aIdEmpregador, Integer aTipoEvento, String aChave, double aDataInicial, double aDataFinal) throws Exception {
        
        ByteBuffer buffer = ByteBuffer.allocate(STR_BUFFER_LEN);
        IntByReference bufferLen = new IntByReference(STR_BUFFER_LEN);
        
        int ret = ACBreSocialLib.INSTANCE.eSocial_ConsultaIdentificadoresEventosTabela(aIdEmpregador, aTipoEvento, aChave, aDataInicial, aDataFinal, buffer, bufferLen);
        checkResult(ret);
        
        return processResult(buffer, bufferLen);
    }
    
    public String consultaIdentificadoresEventosTrabalhador(String aIdEmpregador, String aCPFTrabalhador, double aDataInicial, double aDataFinal)throws Exception {
        
        ByteBuffer buffer = ByteBuffer.allocate(STR_BUFFER_LEN);
        IntByReference bufferLen = new IntByReference(STR_BUFFER_LEN);
        
        int ret = ACBreSocialLib.INSTANCE.eSocial_ConsultaIdentificadoresEventosTrabalhador(aIdEmpregador, aCPFTrabalhador, aDataInicial, aDataFinal, buffer, bufferLen);
        checkResult(ret);
        
        return processResult(buffer, bufferLen);
    }
    
    public String downloadEventos(String aIdEmpregador, String aCPFTrabalhador, double aDataInicial, double aDataFinal) throws Exception {
        
        ByteBuffer buffer = ByteBuffer.allocate(STR_BUFFER_LEN);
        IntByReference bufferLen = new IntByReference(STR_BUFFER_LEN);
        
        int ret = ACBreSocialLib.INSTANCE.eSocial_DownloadEventos(aIdEmpregador, aCPFTrabalhador, aDataInicial, aDataFinal, buffer, bufferLen);
        checkResult(ret);
        
        return processResult(buffer, bufferLen);
    }
    
    public String obterCertificados() throws Exception {
        ByteBuffer buffer = ByteBuffer.allocate(STR_BUFFER_LEN);
        IntByReference bufferLen = new IntByReference(STR_BUFFER_LEN);

        int ret = ACBreSocialLib.INSTANCE.eSocial_ObterCertificados(buffer, bufferLen);
        checkResult(ret);
        return processResult(buffer, bufferLen);
    }
    
    @Override
    protected void UltimoRetorno(ByteBuffer buffer, IntByReference bufferLen) {
        ACBreSocialLib.INSTANCE.eSocial_UltimoRetorno(buffer, bufferLen);
    }
}
